Vi

VakhnenkoIlya opened the pull request.

VakhnenkoIlya committed:
Answer, Qqestion, FakeGenerateTest, MazayTestManeger

Answer, Qqestion Поправил замечания которые были ранее, MazayTestManeger создал интерфейс стартовой страницы, и последующей, FakeGenerateTest создал класс в нем объект interactiveTest

VakhnenkoIlya committed:
ManagerTestsForm

переименовал переменные, из метода
CreateLableTestsWithButtons вынес метод для создания двух кнопок в отдельный метод CreateButtonsStartSettings

VakhnenkoIlya committed:
for comments

roma-outspin commented:
namespace Test_Builder - именование  не совпадает со смыслом. Проект должен называть ёмко и понятно. Это проект для пользователя, в котором он будет управлять своими тестами. КРАТКО, ЕМКО, ПОНЯТНО.

roma-outspin commented:
**class FakeGenerateTest** - именование класса не говорит мне что делает. класс - некая сущность. Логично было бы сделать FakeTestGenerator или нечто подобное
**public Question GetQuestion** - метод который нам не нужен в публичном виде
**public InteractiveTest GetInteractiveTest()** - метод можно назвать короче, в идеале надо было сделать реальную генерацию тестов с разным количеством вопросов и ответов. Для этого можно было использовать математические вопросы (2+2=4 и тп) в это случае метод бы назвать **GenerateTest** и передавать в него количество требуемых вопросов и вариантов ответов.
**public void SerializeTest(InteractiveTest Test)** в именовании метода присутствует слово Тест, в названии проекте слово тест, в названии класса слово тест. Логично было бы оставить просто Serialize, а если ещё постараться то методы сериализации и десериализации вывести в отдельный класс, потому как по смыслу эти методы не нужны в классе с название FakeTestGenerator

roma-outspin commented:
**ManagerTestsForm**
Тут полная беда со стилем кода, всё намешано.

-как минимум наверху должны быть приватные поля (если ты не указываешь идентификатор, то по умолчанию поле private)
все приватные поля начинаем с символа _ (например, _delta)
-Потом конструктор (если он есть)
-затем свойства (это по факту публичные поля пишем их с большой буквы)
-потом публичные методы
-потом приватные методы

private void ManagerTestsForm_Load(object sender, EventArgs e) - если мы не используем метод - удали его
public void CreateButton(int number, string name, int x, int y, EventHandler functionClick) - зачем этот метод публичный? мы же не будем генерировать кнопки для него откуда-то из другого класса

_if (number == 0)
            {
                button.Location = new Point(x, y);
            }
 if (number > 0)
            {
                button.Location = new Point( 95 * number, y);
            }_
два if`а друг за другом, проверяющих одну и ту же переменную, сделай else, чтобы как минимум не проверять второй раз если первая проверка true

private void FolderTest_Click(object sender, EventArgs e) - все понятно, но старайся именовать методы нажатия клавиш более лаконично

 ShowLablesTests, CreateLableTestsWithButtons, CreateButtonsStartSettings, ShowFoldersTests - очень сложно понять смысл из именования этих методов два существительных во множественно числе - сбивают с толку. Переименовать!

roma-outspin commented:
StartForm
всё прекрасно, только не перекидывает на форму, после нажатия кнопки старт, надо перезапускать приложение. Это странное поведение

roma-outspin commented:
**namespace Test_Builder
    static class Program**

Тут поведение слабовато.

            _if (Directory.Exists("Tests"))
            {
                Application.Run(new ManagerTestsForm());
            }
            else Application.Run(new StartForm());_
надо не просто проверять наличие папки Tests, а искать тесты

Далее:
            _FakeGenerateTest fakeTest = new();

_FakeGenerateTest fakeTest = new(); имя переменной fakeTest  не отображается смысл.

roma-outspin commented:
Ну и ещё ManagerTestForm  в обозревателе решения с маленькой буквы начинается
и сериализация/десериализация запускаются только после закрытия формы - странный подход...
Лучше прямо из формы это было тестить.

VakhnenkoIlya committed:
2

Поправил согласно замечаний

roma-outspin commented on MazayTests.Launcher/Test Builder/Deserialization.cs:
```diff
@@ -0,0 +1,21 @@
+    class Deserialization
+    {
+        public InteractiveTest Deserialize()
+        {
```
namespace Test_Builder ?))
Отдельный класс чисто под десериализацию это уже перебор)
File.ReadAllText(@"Tests\Tests1\Test1.json"); 
отдельный класс для десериализации одного конкретного файла? не, так точно не надо делать

roma-outspin commented on MazayTests.Launcher/Test Builder/Program.cs:
```diff
@@ -18,15 +20,16 @@ static void Main(string[] args)
+            else
+            {
+                Application.Run(new StartForm());
+                Application.Run(new ManagerTestsForm());
```
namespace ManagerTests - множественное число
Application.Run(new StartForm());
Application.Run(new ManagerTestsForm());
Логика твоя понятна, но я тебе рассказывал про методы Show и Hide. а это будет работать криво

roma-outspin commented on MazayTests.Launcher/Test Builder/Serialization.cs:
```diff
@@ -0,0 +1,20 @@
+            File.WriteAllText(@"Tests\Tests1\Test1.json", str); 
+        }
+    }
+}
```
тоже самое как  и класc deserialization

roma-outspin committed:
review 01.03

-refactoring
-заново создали проект менеджера тестов
-переопределили логику запуска тестов
-комментарии для будущих правок в файле основной формы
